### DISCUSSION TITLE
为什么看youtube Connection Speed只有2000多，用goproxy都是10000多

### DISCUSSION THREAD

**Discussion by zdzliqian** [SeaHOH/GotoX#29] [问答/咨询] [answered]
阅读之后请删除这些说明文字，然后写下你的问题，谢谢。

提问之前，你可以浏览一下本项目首页的简单说明。
描述清楚问题的情况下标题尽量简洁；
可能的话附上你的系统信息和使用状况；
附上出错时的日志，如果能确认范围就尽量不要全部复制过来，或者截图也可以；
贴文字日志时建议使用以下格式：

```
日志开始
……
日志结束
```

**Comment by jzp820927** [comment]
IP 问题。

**Comment by SeaHOH** [comment]
自带 IP 不保证质量，请自己使用其它工具查找筛选。
涉及到视频下载，请调节 autorange 参数，配置文件注释有详述。
如果 autorange 线程过大，还需调节增大 finder 的保持可用 IP 数量。

**Comment by zdzliqian** [comment]
也就是说把ip.txt里面的ip换了

**Comment by SeaHOH** [comment]
嗯，就是这个理。
我一般最差都在 4000 以上，早上 12000 左右。

**Comment by zdzliqian** [comment]
@SeaHOH  哦 thank you

**Comment by SeaHOH** [comment]
还有要说明一点，youtube Connection Speed 其实是不准确的，一般看 Buffer Health 或自己计算，我上面说的数字都是实际计算得来。

**Comment by zdzliqian** [comment]
怎么算？

**Comment by SeaHOH** [comment]
找一段大小合适的视频下载，看下载日志开始结束时间和下载大小。

**Comment by zdzliqian** [comment]
还可以下载？

**Comment by SeaHOH** [comment]
当然，怎么下载我就不管了。你直接看网页视频，它是也会下载的。

**Comment by caofengzi** [comment]
下載速度應該取決於本地網絡，俺這裡下載youtude視頻是1-3m/s。

**Comment by SeaHOH** [comment, marked as answer]
@zdzliqian 
a9869c0fcfca8db052513f498d02b3289eb0d7e1 这个 commits 之后可以达到与 GoProxy 相同的效果。原理是增加 TCP 最大接收窗口，U2B 的 Connection Speed 测的是瞬时速度（带宽），所以数值会提高。
由于 AppID 的带宽是有限制的，所以这个修改**并不会增加总的带宽**，其好处是可以减少使用中的链接被干扰的机率（在干扰之前就传输完毕），但是会增加内存使用（不表现在 Python 而是系统 TCP/IP 协议栈）。
